### ISSUE TITLE
Gracile is incompatible with typescript aliases

### ISSUE THREAD

**Issue by lschierer** [gracile-web/gracile#3]
I tried to set up aliases in my tsconfig and the vite section of the gracile.config.ts files to make including the documents easier from routes at variable directory depths, and to my surprise, it blew up in my face.

I set up a minimal reproduction [here](https://github.com/lschierer/gracile_demos/tree/typescript_aliases).  If you check that out and run the command ```pnpm build``` you will see that it produces the error

```
12:21:37 AM [vite] Pre-transform error: Failed to load url components/testComponent (resolved id: components/testComponent) in /Users/luke/src/aliasesDemo/src/document.ts. Does the file exist?
12:21:37 AM [vite] Error when evaluating SSR module /src/document.ts: failed to import "components/testComponent"
|- Error: Cannot find module 'components/testComponent' imported from '/Users/luke/src/aliasesDemo/src/document.ts'
....
```

however ```pnpm tsc``` does not complain.  I have tried this from memory of how to do so, then when that didn't work, following the directions at both

<https://dev.to/larswaechter/path-aliases-with-typescript-in-nodejs-4353> and <https://www.delasign.com/blog/react-vitejs-typescript-path-alias/>  which differ slightly (I tried each way) both from each other and from what I've done in prior projects.  

Either way, this ought to work and would make maintaining the route files considerably cleaner. 

Thanks!!

**Comment by JulianCataldo** [comment]
Hi!

Have you tried https://www.npmjs.com/package/vite-tsconfig-paths ?

That's what Remix is using.

**Comment by lschierer** [comment]
that got me close, I ended up adding --root . to several of the commands as well, see [this commit](https://github.com/lschierer/gracile_demos/commit/ad34e301bd06839166d90c82d4b1567c8bed07aa) if curious.

**Comment by JulianCataldo** [comment]
I tried https://nodejs.org/api/packages.html#subpath-imports they work well with the latest typescript and Node.
They work with Vite, too, but only for direct file-to-file mapping it seems, no glob.

Gracile is now a Vite plugin, meaning the Alias thing is just the same as every framework/bare Vite users are facing.

One could source a Vite plugin for doing this. BTW. I think imports maps are better than TS paths.

Closing as it's way out of scope.
